### PR TITLE
Adds the 84mm solid rocket booster.

### PR DESCRIPTION
**Pull request by AcapitalA**
What's it do: Adds an 84mm solid rocket booster to the nuclear uplink.
Yeah, but like, what's it good for: You can cram that shit into a rocket launcher and fire it.
YES BUT DOES IT EXPLODE: It fucking sears everything in the backblast for a SERIOUS amount of burn!
Okay, sounds pretty fuckin neato but the projectile itself: Simple canister flies, explodes and sets people on fire.
So how much does it cost: A single fucking TC, seriously.
Won't that shit be OP: You can fucking end 15 people average with a single L6 mag, i'm pretty sure it won't be OP, just on-par with actually good guns.

Does it do anything unique?
Yeah, it turns the rocket launcher into a short ranged flamethrower for a short while.

So, why?
It's fun, that's why.

### CODE REVIEW

**Comment by AcapitalA** [comment]
Since bots got blacklisted from service, someone might wanna put this PR on hold until it gets fixed...

File: code/modules/projectiles/guns/ballistic.dm [hunk 1]
```diff
@@ -47,6 +47,8 @@
 
 	///Whether the gun will spawn loaded with a magazine
 	var/spawnwithmagazine = TRUE
+	///If the weapon fires flames out of it's back
+	var/backblast = FALSE
```
Comment (TheChosenEvilOne) [inline code/modules/projectiles/guns/ballistic.dm:51]: change this to a variable that is either null or the type of the backblast bullet and set it in .proc/chamber_round according to the bullet loaded, maybe add a var to bullets for this as well.

File: code/datums/elements/backblast.dm [hunk 2]
```diff
@@ -64,7 +66,12 @@
 /// For firing an actual backblast pellet
 /datum/element/backblast/proc/pew(turf/target_turf, obj/item/gun/weapon, mob/living/user)
 	//Shooting Code:
-	var/obj/projectile/bullet/incendiary/backblast/P = new (get_turf(user))
+	backblast_type = /obj/projectile/bullet/incendiary/backblast
+
+	if(locate(/obj/item/ammo_casing/caseless/rocket/solidfuel) in weapon)
+		backblast_type = /obj/projectile/bullet/incendiary/backblast/solidfuel
+
+	var/obj/projectile/bullet/incendiary/backblast/P = new backblast_type(get_turf(user))
```
Comment (TheChosenEvilOne) [inline code/datums/elements/backblast.dm]: make this check if weapon is ballistic and then use the weapon's backblast type
something like:
```
var/obj/projectile/bullet/P
if (istype(weapon, /obj/item/gun/ballistic))
      var/obj/item/gun/ballistic/W = weapon
      P = new W.backblast(get_turf(user))
else
      P = new /obj/projectile/bullet/incendiary/backblast(get_turf(user))
```

File: code/datums/elements/backblast.dm [hunk 3]
```diff
@@ -14,6 +14,8 @@
 	var/angle_spread
 	/// How far each plume of fire will fly, assuming it doesn't hit a mob
 	var/range
+	/// What kind of backblast projectile will be fired.
+	var/backblast_type
```
Comment (TheChosenEvilOne) [inline code/datums/elements/backblast.dm]: this is not needed

**Comment by TheChosenEvilOne** [comment]
> Since bots got blacklisted from service, someone might wanna put this PR on hold until it gets fixed...

it is just icondiff and mapdiff, it is fine.

File: code/modules/projectiles/guns/ballistic.dm (hunk 1 above)
Comment (AcapitalA) [reply code/modules/projectiles/guns/ballistic.dm:51]: Did this along with all the other changes. Gives me the error that the var doesn't have a definition tied to it. Might have to do something with the other changes i made.

File: code/datums/elements/backblast.dm [hunk 4]
```diff
@@ -64,7 +64,12 @@
 /// For firing an actual backblast pellet
 /datum/element/backblast/proc/pew(turf/target_turf, obj/item/gun/weapon, mob/living/user)
 	//Shooting Code:
-	var/obj/projectile/bullet/incendiary/backblast/P = new (get_turf(user))
+	var/obj/projectile/bullet/P
+	if (istype(weapon, /obj/item/gun/ballistic))
+		var/obj/item/gun/ballistic/W = weapon
+		P = new W.chambered.backblast_type(get_turf(user))
+	else
+		P = new /obj/projectile/bullet/incendiary/backblast(get_turf(user))
```
Comment (TheChosenEvilOne) [inline code/datums/elements/backblast.dm]: all guns have the chambered var
```suggestion
	if (weapon.chambered.backblast_type)
		P = new weapon.chambered.backblast_type(get_turf(user))
	else
		P = new /obj/projectile/bullet/incendiary/backblast(get_turf(user))
```

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 5]
```diff
@@ -70,6 +70,30 @@
...
+	icon_state = "solidfuel_syndie"
+	damage = 20 //still hurts to get smacked by a flaming canister at high speeds
+	dismemberment = 0
+	embedding = null
+	shrapnel_type = null
+	var/fire_stacks = 6
+
+/obj/projectile/bullet/a84mm/solidfuel/on_hit(atom/target, blocked = 0)
+	if(iscarbon(target))
+		if(isliving(target))
```
Comment (TheChosenEvilOne) [inline code/modules/projectiles/projectile/special/rocket.dm]: carbon is living (/mob/living/carbon) isliving doesn't check if the target is living but if the target is /mob/living.
you probably want to drop the isliving check completely.

**Pull request merged**